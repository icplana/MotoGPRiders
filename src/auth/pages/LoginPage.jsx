import React, { useContext, useRef } from 'react'
import { useForm } from '../../hooks/useForm'
import { AuthContext } from '../context'
import { Link, useNavigate } from 'react-router-dom'
import { signInEmail, signInWithGoogle } from '../../Firebase/firebase'

export const LoginPage = () => {

  const { login } = useContext( AuthContext )

  const wrongAlert = useRef()
  const wrong2Alert = useRef()

  const onLoginGoogle = async (e) => {
    e.preventDefault()

    const user = await signInWithGoogle()
    
    if ( user.auth ) {
      
      login( user.email )
      return
    }

    wrong2Alert.current.classList.remove('d-none')

    setTimeout(() => {
      wrong2Alert.current.classList.add('d-none')
    }, 5000);
  }

  const onLogin = async (e) => {
    e.preventDefault()
    const user = await signInEmail( email, password )

    if ( user.auth ) {
      
      login( user.email )
      return
    }

    wrongAlert.current.classList.remove('d-none')

    setTimeout(() => {
      wrongAlert.current.classList.add('d-none')
    }, 5000);

    
  }

  const { onInputChange, email, password } = useForm({ email: '', password: '' })
  return (
    <div className='bg-light py-5 min-vh-100'>
      <form className='mx-5 bg-primary text-white rounded-3 p-4' onSubmit={ onLogin }>
      <h3 className='mb-4'>Login</h3>
      
        <div className=' mb-3'>          
          <label htmlFor="email" className='form-label'>Correo</label>
          <input 
            type="email" 
            id='email'
            name='email' 
            className='form-control' 
            placeholder='correo@correo.com'
            value={ email }
            onInput={ onInputChange }            
          />
        </div>
        <div className='mb-3'>
          <label className='form-label'>Contraseña</label>
          <input 
            type="password" 
            className='form-control' 
            placeholder='********'
            name='password'
            value={ password }
            onInput={ onInputChange }  
          />
        </div>

        <button className='btn btn-secondary' onClick={ onLogin } >
          Login
        </button>

        <button className='btn btn-secondary ms-5' onClick={ onLoginGoogle }>
          Login with google
        </button>

        <div className="alert alert-danger d-none mt-3" role="alert" ref={ wrongAlert }>
            Usuario y/o contraseña incorrecto/s.
        </div>

        <div className="alert alert-danger d-none mt-3" role="alert" ref={ wrong2Alert }>
            Algo salio mal.
        </div>


      </form>
        <Link to="/register" className='ms-5 btn btn-primary mt-4'>Crear una cuenta!</Link>
    </div>
  )
}
