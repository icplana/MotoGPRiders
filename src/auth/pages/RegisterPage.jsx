import { useContext, useRef } from "react"
import { useForm } from "../../hooks/useForm"
import { registerEmail } from "../../Firebase/firebase"
import { Link, Navigate } from "react-router-dom"
import { AuthContext } from "../context/AuthContext"



export const RegisterPage = () => {
    
    const onRegister = async (e) => {
        e.preventDefault()
        let validation = true
        
        if ( nombre.length < 1 ) {
            validation = false
            nameAlert.current.classList.remove('d-none')
        }
        if ( password.length < 6 ) {
            validation = false
            passwordAlert.current.classList.remove('d-none')
        }
        if ( email.length < 1 || !email.includes('@')) {
            validation = false
            correoAlert.current.classList.remove('d-none')
        }


        if ( validation === true ) {
            
           const resp = await registerEmail( email, password )
           if ( resp.auth ) {
                succesAlert.current.classList.remove('d-none')
           }
           

            
        }
    }
    
    const handleInputChange = (e) => {
        onInputChange(e)
        
        if ( nombre.length > 1 ) nameAlert.current.classList.add('d-none')
        if ( password.length > 4 ) passwordAlert.current.classList.add('d-none')
        if ( email.length > 1 && email.includes('@')) correoAlert.current.classList.add('d-none')
    
    }

    const nameAlert = useRef()
    const correoAlert = useRef()
    const passwordAlert = useRef()
    const succesAlert = useRef()
   

    const { onInputChange, email, password, nombre } = useForm({ email: '', password: '' , nombre: '' })

    const { login } = useContext( AuthContext )

  return (
    <div className='bg-light py-5 min-vh-100'>
    <form className='mx-5 bg-primary text-white rounded-3 p-4' onSubmit={onRegister}>
      <h3 className='mb-4'>Registro de cuenta</h3>
      
        <div className=' mb-3'>          
          <label htmlFor="nombre" className='form-label'>Nombre</label>
          <input 
            type="text" 
            id='nombre'
            name='nombre' 
            className='form-control mb-2' 
            placeholder='Juan Hidalgo'
            value={ nombre }
            onInput={ onInputChange }            
          />
          <div className="alert alert-danger d-none" role="alert" ref={nameAlert}>
            Nombre no v치lido
          </div>
        </div>
      
        <div className=' mb-3'>          
          <label htmlFor="email" className='form-label'>Correo</label>
          <input 
            type="email" 
            id='email'
            name='email' 
            className='form-control mb-2' 
            placeholder='correo@correo.com'
            value={ email }
            onInput={ handleInputChange }            
          />
           <div className="alert alert-danger d-none" role="alert" ref={correoAlert}>
            Email no v치lido
          </div>
        </div>


        <div className='mb-3'>
          <label className='form-label'>Contrase침a</label>
          <input 
            type="password" 
            className='form-control mb-2' 
            placeholder='********'
            name='password'
            value={ password }
            onInput={ handleInputChange }  
          />
           <div className="alert alert-danger d-none" role="alert" ref={passwordAlert}>
            La contrase침a debe tener almenos 6 caracteres
          </div>
        </div>

        <button className='btn btn-secondary' >
          Registrarse
        </button>

        <div className="alert alert-success d-none mt-2" role="alert" ref={succesAlert}>
            Registro Correcto

            <Link to="/motogp" className="ms-3 btn btn-success" onClick={ () => login( email )}>Entrar</Link>
        </div>



      </form>

      <Link to="/login" className="btn btn-primary ms-5 mt-3">Ya tengo cuenta!</Link>
    </div>
  )
}
