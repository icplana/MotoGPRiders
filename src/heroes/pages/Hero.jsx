import { Navigate, useNavigate, useParams } from "react-router-dom"
import { getHeroById } from "../helpers"
import { useMemo } from "react"

export const Hero = () => {

  const { heroId } = useParams() 
 
  const actualHero = useMemo( () => getHeroById( heroId ), [ heroId ] )

  const navigate = useNavigate()

  const onNavigateBack = () => {
    navigate(-1)
  }

  
  if ( !actualHero ){
    return <Navigate to="/dc" />
  }


  return (
    <div className="row mt-5">
      <div className="col-4">
        <img 
          src= { `/heroes/${ heroId }.jpg` }
          alt={ actualHero.superhero } 
          className="img-thumbnail animate__animated animate__fadeInLeft"
        />
      </div>

      <div className="col-8">
        <h3>{ actualHero.superhero }</h3>
        <ul className="list-group list-group-flush">
          <li className="list-group-item"><b>Alter ego:</b> { actualHero.alter_ego}</li>
          <li className="list-group-item"><b>Publisher:</b> { actualHero.publisher}</li>
          <li className="list-group-item"><b>First appearance:</b> { actualHero.first_appearance}</li>
        </ul>


        <h5 className=" mt3">Characters</h5>
        <p>{ actualHero.characters }</p>


        <button
          className="btn btn-outline-primary"
          onClick={ onNavigateBack }
        >Regresar</button>
      </div>


    </div>
  )
}
