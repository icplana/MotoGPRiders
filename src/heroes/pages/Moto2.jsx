import { RiderList } from "../components/RiderList"




export const Moto2 = () => {
  return (
    <>
      <h1>Moto 2</h1>
      <hr />

      <RiderList league={ 'Moto2'} />
    </>
  )
}
