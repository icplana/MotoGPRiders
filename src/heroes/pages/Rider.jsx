import { Navigate, useNavigate, useParams } from "react-router-dom"
import { getRiderById } from "../helpers"
import { useMemo } from "react"

export const Rider = () => {

  const { riderId } = useParams() 
 
  const actualRider = useMemo( () => getRiderById( riderId ), [ riderId ] )

  const navigate = useNavigate()

  const onNavigateBack = () => {
    navigate(-1)
  }

  
  if ( !actualRider ){
    return <Navigate to="/dc" />
  }


  return (
    <div className="row mt-5">
      <div className="col-4">
        <img 
          src= { actualRider.imgURL }
          alt={ actualRider.name } 
          className="img-thumbnail animate__animated animate__fadeInLeft"
        />
      </div>

      <div className="col-8">
        <h3>{ actualRider.name }</h3>       

        <button
          className="btn btn-outline-primary"
          onClick={ onNavigateBack }
        >Regresar</button>
      </div>


    </div>
  )
}
