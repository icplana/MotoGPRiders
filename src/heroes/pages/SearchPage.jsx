import { useLocation, useNavigate } from "react-router-dom"
import queryString from "query-string"
import { useForm } from "../../hooks/useForm"
import { HeroCard } from "../components/HeroCard"
import { useState } from "react"
import { getHerosByName } from "../helpers"


export const SearchPage = () => {

  
  const navigate = useNavigate()

  const location = useLocation()

  // const query = queryString.parse( location.search )
  // console.log(query)

  const { q: searchValue = '' } = queryString.parse( location.search )

  const results = getHerosByName( searchValue ) 
  

  const { searchText, onInputChange, onResetForm } = useForm({
    searchText: ''
  })

  const onSearchSubmit = e => {
    e.preventDefault()

    if ( searchText.trim().length < 1 ) return
    
   

    navigate(`?q=${ searchText.toLowerCase().trim() }`)
    
  }

  return (
    <>
      <h1>Search</h1>
      <hr />
      <div className="row">

        <div className="col-5">
          <form className="form-inline mt-2" onSubmit={ onSearchSubmit }>
            <input 
              className="form-control" 
              type="text" 
              placeholder="Search" 
              name="searchText"
              autoComplete="off"
              value={ searchText }
              onChange={ onInputChange }
            />
            <button className="btn btn-outline-success mt-2" type="submit">Search</button>
          </form>
        </div>

        <div className="col-7">
          <h4>Results</h4>
          <hr />

          <div className={`alert alert-primary ${(searchValue) ? 'd-none' :''}`}>Search a hero</div>

          <div className={`alert alert-danger ${(results.length === 0 ) ? '' :'d-none'}`}>No results for <b>{ searchValue }</b></div>

          <div className={`${(searchValue) ? '' :'d-none'}`}>
          {
            results.map( hero =>(
              <HeroCard key={ hero.id} hero={hero} />
            ))
          }      
          </div>
        </div>

      </div>
    </>
  )
}
