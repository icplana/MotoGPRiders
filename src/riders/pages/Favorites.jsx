import React from 'react'
import { RiderList } from '../components/RiderList'

export const Favorites = () => {
  return (
    <>
      <h1>Favoritos</h1>
      <hr />

      <RiderList league={ 'MotoGP' } />
    </>
  )
}
