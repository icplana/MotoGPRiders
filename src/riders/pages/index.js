

export * from './MotoGP'
export * from './Moto2'
export * from './Rider'
