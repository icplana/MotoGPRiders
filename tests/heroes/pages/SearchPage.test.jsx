import { render, screen } from "@testing-library/react";
import { SearchPage } from "../../../src/heroes/pages/SearchPage";
import { MemoryRouter } from "react-router-dom";




describe('Pruebas sobre SearchPage.jsx', () => { 

    test('debe de mostrarse correctamente con valores por defecto', () => {
        

        const { container } = render( 
            <MemoryRouter>
                <SearchPage />
            </MemoryRouter>
        )
        

        expect( container ).toMatchSnapshot()
    });


    test('debe de mostrar a batman i input con el valor del querystring', () => {
        render(
            <MemoryRouter initialEntries={['/search?q=batman']}>
                <SearchPage/>
            </MemoryRouter>
        )
        

            expect( screen.getByText('Batman') ).toBeTruthy()
        
        expect( screen.getByText('Search a hero').classList.contains('d-none') ).toBeTruthy()
        expect( screen.getByText('Search a hero').nextSibling.classList.contains('d-none') ).toBeTruthy()
        console.log(screen.getByText('Search a hero').nextSibling.style)
        // expect( screen.getByText('Search a hero').nextSibling.style.display ).toBe('none')
        // no aparece display: none en los styles al pasarlos por consola, ni pasa los tests pero no se muestra y el elemento si que tiene la clase d-none ademas en chrome tools el elemento muestra display: none!important





        // esta prueba no funciona pq el codigo no contempla enviar la informaci√≥n del query(url) al input, solo en sentido contrario ( en el codigo de fernando si se contempla )       
        // expect( screen.getByRole('textbox').value.trim().toLowerCase()).toBe('batman')
    });
 })