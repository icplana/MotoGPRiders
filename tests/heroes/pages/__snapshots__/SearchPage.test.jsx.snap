// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre SearchPage.jsx debe de mostrarse correctamente con valores por defecto 1`] = `
<div>
  <h1>
    Search
  </h1>
  <hr />
  <div
    class="row"
  >
    <div
      class="col-5"
    >
      <form
        class="form-inline mt-2"
      >
        <input
          autocomplete="off"
          class="form-control"
          name="searchText"
          placeholder="Search"
          type="text"
          value=""
        />
        <button
          class="btn btn-outline-success mt-2"
          type="submit"
        >
          Search
        </button>
      </form>
    </div>
    <div
      class="col-7"
    >
      <h4>
        Results
      </h4>
      <hr />
      <div
        class="alert alert-primary "
      >
        Search a hero
      </div>
      <div
        class="alert alert-danger d-none"
      >
        No results for 
        <b />
      </div>
      <div
        class="d-none"
      >
        <div
          class="cardBox card p-2 mb-2"
        >
          <img
            alt="Batman"
            class="animate__animated animate__fadeIn imgCard mx-auto"
            src="heroes/dc-batman.jpg"
          />
          <h3>
            Batman
          </h3>
          <h5>
            Bruce Wayne
          </h5>
          <h5 />
          <p>
            Detective Comics #27
          </p>
          <a
            class="btn btn-secondary mt-auto"
            href="/hero/dc-batman"
          >
            Más información...
          </a>
        </div>
        <div
          class="cardBox card p-2 mb-2"
        >
          <img
            alt="Superman"
            class="animate__animated animate__fadeIn imgCard mx-auto"
            src="heroes/dc-superman.jpg"
          />
          <h3>
            Superman
          </h3>
          <h5>
            Kal-El
          </h5>
          <h5 />
          <p>
            Action Comics #1
          </p>
          <a
            class="btn btn-secondary mt-auto"
            href="/hero/dc-superman"
          >
            Más información...
          </a>
        </div>
        <div
          class="cardBox card p-2 mb-2"
        >
          <img
            alt="Flash"
            class="animate__animated animate__fadeIn imgCard mx-auto"
            src="heroes/dc-flash.jpg"
          />
          <h3>
            Flash
          </h3>
          <h5>
            Jay Garrick
          </h5>
          <h5>
            Jay Garrick, Barry Allen, Wally West, Bart Allen
          </h5>
          <p>
            Flash Comics #1
          </p>
          <a
            class="btn btn-secondary mt-auto"
            href="/hero/dc-flash"
          >
            Más información...
          </a>
        </div>
        <div
          class="cardBox card p-2 mb-2"
        >
          <img
            alt="Green Lantern"
            class="animate__animated animate__fadeIn imgCard mx-auto"
            src="heroes/dc-green.jpg"
          />
          <h3>
            Green Lantern
          </h3>
          <h5>
            Alan Scott
          </h5>
          <h5>
            Alan Scott, Hal Jordan, Guy Gardner, John Stewart, Kyle Raynor, Jade, Sinestro, Simon Baz
          </h5>
          <p>
            All-American Comics #16
          </p>
          <a
            class="btn btn-secondary mt-auto"
            href="/hero/dc-green"
          >
            Más información...
          </a>
        </div>
        <div
          class="cardBox card p-2 mb-2"
        >
          <img
            alt="Green Arrow"
            class="animate__animated animate__fadeIn imgCard mx-auto"
            src="heroes/dc-arrow.jpg"
          />
          <h3>
            Green Arrow
          </h3>
          <h5>
            Oliver Queen
          </h5>
          <h5 />
          <p>
            More Fun Comics #73
          </p>
          <a
            class="btn btn-secondary mt-auto"
            href="/hero/dc-arrow"
          >
            Más información...
          </a>
        </div>
        <div
          class="cardBox card p-2 mb-2"
        >
          <img
            alt="Wonder Woman"
            class="animate__animated animate__fadeIn imgCard mx-auto"
            src="heroes/dc-wonder.jpg"
          />
          <h3>
            Wonder Woman
          </h3>
          <h5>
            Princess Diana
          </h5>
          <h5 />
          <p>
            All Star Comics #8
          </p>
          <a
            class="btn btn-secondary mt-auto"
            href="/hero/dc-wonder"
          >
            Más información...
          </a>
        </div>
        <div
          class="cardBox card p-2 mb-2"
        >
          <img
            alt="Martian Manhunter"
            class="animate__animated animate__fadeIn imgCard mx-auto"
            src="heroes/dc-martian.jpg"
          />
          <h3>
            Martian Manhunter
          </h3>
          <h5>
            J'onn J'onzz
          </h5>
          <h5>
            Martian Manhunter
          </h5>
          <p>
            Detective Comics #225
          </p>
          <a
            class="btn btn-secondary mt-auto"
            href="/hero/dc-martian"
          >
            Más información...
          </a>
        </div>
        <div
          class="cardBox card p-2 mb-2"
        >
          <img
            alt="Robin/Nightwing"
            class="animate__animated animate__fadeIn imgCard mx-auto"
            src="heroes/dc-robin.jpg"
          />
          <h3>
            Robin/Nightwing
          </h3>
          <h5>
            Dick Grayson
          </h5>
          <h5 />
          <p>
            Detective Comics #38
          </p>
          <a
            class="btn btn-secondary mt-auto"
            href="/hero/dc-robin"
          >
            Más información...
          </a>
        </div>
        <div
          class="cardBox card p-2 mb-2"
        >
          <img
            alt="Blue Beetle"
            class="animate__animated animate__fadeIn imgCard mx-auto"
            src="heroes/dc-blue.jpg"
          />
          <h3>
            Blue Beetle
          </h3>
          <h5>
            Dan Garret
          </h5>
          <h5>
            Dan Garret, Ted Kord, Jaime Reyes
          </h5>
          <p>
            Mystery Men Comics #1
          </p>
          <a
            class="btn btn-secondary mt-auto"
            href="/hero/dc-blue"
          >
            Más información...
          </a>
        </div>
        <div
          class="cardBox card p-2 mb-2"
        >
          <img
            alt="Black Canary"
            class="animate__animated animate__fadeIn imgCard mx-auto"
            src="heroes/dc-black.jpg"
          />
          <h3>
            Black Canary
          </h3>
          <h5>
            Dinah Drake
          </h5>
          <h5>
            Dinah Drake, Dinah Lance
          </h5>
          <p>
            Flash Comics #86
          </p>
          <a
            class="btn btn-secondary mt-auto"
            href="/hero/dc-black"
          >
            Más información...
          </a>
        </div>
        <div
          class="cardBox card p-2 mb-2"
        >
          <img
            alt="Spider Man"
            class="animate__animated animate__fadeIn imgCard mx-auto"
            src="heroes/marvel-spider.jpg"
          />
          <h3>
            Spider Man
          </h3>
          <h5>
            Peter Parker
          </h5>
          <h5 />
          <p>
            Amazing Fantasy #15
          </p>
          <a
            class="btn btn-secondary mt-auto"
            href="/hero/marvel-spider"
          >
            Más información...
          </a>
        </div>
        <div
          class="cardBox card p-2 mb-2"
        >
          <img
            alt="Captain America"
            class="animate__animated animate__fadeIn imgCard mx-auto"
            src="heroes/marvel-captain.jpg"
          />
          <h3>
            Captain America
          </h3>
          <h5>
            Steve Rogers
          </h5>
          <h5 />
          <p>
            Captain America Comics #1
          </p>
          <a
            class="btn btn-secondary mt-auto"
            href="/hero/marvel-captain"
          >
            Más información...
          </a>
        </div>
        <div
          class="cardBox card p-2 mb-2"
        >
          <img
            alt="Iron Man"
            class="animate__animated animate__fadeIn imgCard mx-auto"
            src="heroes/marvel-iron.jpg"
          />
          <h3>
            Iron Man
          </h3>
          <h5>
            Tony Stark
          </h5>
          <h5 />
          <p>
            Tales of Suspense #39
          </p>
          <a
            class="btn btn-secondary mt-auto"
            href="/hero/marvel-iron"
          >
            Más información...
          </a>
        </div>
        <div
          class="cardBox card p-2 mb-2"
        >
          <img
            alt="Thor"
            class="animate__animated animate__fadeIn imgCard mx-auto"
            src="heroes/marvel-thor.jpg"
          />
          <h3>
            Thor
          </h3>
          <h5>
            Thor Odinson
          </h5>
          <h5 />
          <p>
            Journey into Myster #83
          </p>
          <a
            class="btn btn-secondary mt-auto"
            href="/hero/marvel-thor"
          >
            Más información...
          </a>
        </div>
        <div
          class="cardBox card p-2 mb-2"
        >
          <img
            alt="Hulk"
            class="animate__animated animate__fadeIn imgCard mx-auto"
            src="heroes/marvel-hulk.jpg"
          />
          <h3>
            Hulk
          </h3>
          <h5>
            Bruce Banner
          </h5>
          <h5 />
          <p>
            The Incredible Hulk #1
          </p>
          <a
            class="btn btn-secondary mt-auto"
            href="/hero/marvel-hulk"
          >
            Más información...
          </a>
        </div>
        <div
          class="cardBox card p-2 mb-2"
        >
          <img
            alt="Wolverine"
            class="animate__animated animate__fadeIn imgCard mx-auto"
            src="heroes/marvel-wolverine.jpg"
          />
          <h3>
            Wolverine
          </h3>
          <h5>
            James Howlett
          </h5>
          <h5 />
          <p>
            The Incredible Hulk #180
          </p>
          <a
            class="btn btn-secondary mt-auto"
            href="/hero/marvel-wolverine"
          >
            Más información...
          </a>
        </div>
        <div
          class="cardBox card p-2 mb-2"
        >
          <img
            alt="Daredevil"
            class="animate__animated animate__fadeIn imgCard mx-auto"
            src="heroes/marvel-daredevil.jpg"
          />
          <h3>
            Daredevil
          </h3>
          <h5>
            Matthew Michael Murdock
          </h5>
          <h5 />
          <p>
            Daredevil #1
          </p>
          <a
            class="btn btn-secondary mt-auto"
            href="/hero/marvel-daredevil"
          >
            Más información...
          </a>
        </div>
        <div
          class="cardBox card p-2 mb-2"
        >
          <img
            alt="Hawkeye"
            class="animate__animated animate__fadeIn imgCard mx-auto"
            src="heroes/marvel-hawkeye.jpg"
          />
          <h3>
            Hawkeye
          </h3>
          <h5>
            Clinton Francis Barton
          </h5>
          <h5 />
          <p>
            Tales of Suspense #57
          </p>
          <a
            class="btn btn-secondary mt-auto"
            href="/hero/marvel-hawkeye"
          >
            Más información...
          </a>
        </div>
        <div
          class="cardBox card p-2 mb-2"
        >
          <img
            alt="Cyclops"
            class="animate__animated animate__fadeIn imgCard mx-auto"
            src="heroes/marvel-cyclops.jpg"
          />
          <h3>
            Cyclops
          </h3>
          <h5>
            Scott Summers
          </h5>
          <h5 />
          <p>
            X-Men #1
          </p>
          <a
            class="btn btn-secondary mt-auto"
            href="/hero/marvel-cyclops"
          >
            Más información...
          </a>
        </div>
        <div
          class="cardBox card p-2 mb-2"
        >
          <img
            alt="Silver Surfer"
            class="animate__animated animate__fadeIn imgCard mx-auto"
            src="heroes/marvel-silver.jpg"
          />
          <h3>
            Silver Surfer
          </h3>
          <h5>
            Norrin Radd
          </h5>
          <h5 />
          <p>
            The Fantastic Four #48
          </p>
          <a
            class="btn btn-secondary mt-auto"
            href="/hero/marvel-silver"
          >
            Más información...
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
